{"metadata": {"kernelspec": {"name": "python3", "display_name": "Python 3.11", "language": "python"}, "language_info": {"name": "python", "version": "3.11.13", "mimetype": "text/x-python", "codemirror_mode": {"name": "ipython", "version": 3}, "pygments_lexer": "ipython3", "nbconvert_exporter": "python", "file_extension": ".py"}}, "nbformat_minor": 4, "nbformat": 4, "cells": [{"cell_type": "code", "source": "!pip install --upgrade ibm-watsonx-ai\nfrom ibm_watsonx_ai.foundation_models import Model\nparameters = {\n    \"decoding_method\": \"greedy\",\n    \"max_new_tokens\": 100\n}\n\nmodel = Model(\n    model_id=\"granite-13b-instruct-v2\",\n    params=parameters,\n    project_id=\"ce7c019d-e903-4766-aee3-b3e3c15574c9\"\n)\nprompt = \"Explain the concept of AI to a 5-year-old kid.\"\n\nresponse = model.generate(prompt)\nprint(\"Generated Text:\", response)", "metadata": {"id": "af9fe7dc-7319-4a93-b390-a71cf987c99c"}, "outputs": [{"name": "stdout", "text": "Requirement already satisfied: ibm-watsonx-ai in /opt/conda/envs/Python-RT24.1/lib/python3.11/site-packages (1.3.26)\nRequirement already satisfied: requests in /opt/conda/envs/Python-RT24.1/lib/python3.11/site-packages (from ibm-watsonx-ai) (2.32.2)\nRequirement already satisfied: httpx<0.29,>=0.27 in /opt/conda/envs/Python-RT24.1/lib/python3.11/site-packages (from ibm-watsonx-ai) (0.28.1)\nRequirement already satisfied: urllib3 in /opt/conda/envs/Python-RT24.1/lib/python3.11/site-packages (from ibm-watsonx-ai) (1.26.19)\nRequirement already satisfied: pandas<2.3.0,>=0.24.2 in /opt/conda/envs/Python-RT24.1/lib/python3.11/site-packages (from ibm-watsonx-ai) (2.1.4)\nRequirement already satisfied: certifi in /opt/conda/envs/Python-RT24.1/lib/python3.11/site-packages (from ibm-watsonx-ai) (2025.4.26)\nRequirement already satisfied: lomond in /opt/conda/envs/Python-RT24.1/lib/python3.11/site-packages (from ibm-watsonx-ai) (0.3.3)\nRequirement already satisfied: tabulate in /opt/conda/envs/Python-RT24.1/lib/python3.11/site-packages (from ibm-watsonx-ai) (0.8.10)\nRequirement already satisfied: packaging in /opt/conda/envs/Python-RT24.1/lib/python3.11/site-packages (from ibm-watsonx-ai) (23.2)\nRequirement already satisfied: ibm-cos-sdk<2.15.0,>=2.12.0 in /opt/conda/envs/Python-RT24.1/lib/python3.11/site-packages (from ibm-watsonx-ai) (2.13.4)\nRequirement already satisfied: anyio in /opt/conda/envs/Python-RT24.1/lib/python3.11/site-packages (from httpx<0.29,>=0.27->ibm-watsonx-ai) (3.5.0)\nRequirement already satisfied: httpcore==1.* in /opt/conda/envs/Python-RT24.1/lib/python3.11/site-packages (from httpx<0.29,>=0.27->ibm-watsonx-ai) (1.0.9)\nRequirement already satisfied: idna in /opt/conda/envs/Python-RT24.1/lib/python3.11/site-packages (from httpx<0.29,>=0.27->ibm-watsonx-ai) (3.7)\nRequirement already satisfied: h11>=0.16 in /opt/conda/envs/Python-RT24.1/lib/python3.11/site-packages (from httpcore==1.*->httpx<0.29,>=0.27->ibm-watsonx-ai) (0.16.0)\nRequirement already satisfied: ibm-cos-sdk-core==2.13.4 in /opt/conda/envs/Python-RT24.1/lib/python3.11/site-packages (from ibm-cos-sdk<2.15.0,>=2.12.0->ibm-watsonx-ai) (2.13.4)\nRequirement already satisfied: ibm-cos-sdk-s3transfer==2.13.4 in /opt/conda/envs/Python-RT24.1/lib/python3.11/site-packages (from ibm-cos-sdk<2.15.0,>=2.12.0->ibm-watsonx-ai) (2.13.4)\nRequirement already satisfied: jmespath<=1.0.1,>=0.10.0 in /opt/conda/envs/Python-RT24.1/lib/python3.11/site-packages (from ibm-cos-sdk<2.15.0,>=2.12.0->ibm-watsonx-ai) (1.0.1)\nRequirement already satisfied: python-dateutil<3.0.0,>=2.8.2 in /opt/conda/envs/Python-RT24.1/lib/python3.11/site-packages (from ibm-cos-sdk-core==2.13.4->ibm-cos-sdk<2.15.0,>=2.12.0->ibm-watsonx-ai) (2.8.2)\nRequirement already satisfied: numpy<2,>=1.23.2 in /opt/conda/envs/Python-RT24.1/lib/python3.11/site-packages (from pandas<2.3.0,>=0.24.2->ibm-watsonx-ai) (1.26.4)\nRequirement already satisfied: pytz>=2020.1 in /opt/conda/envs/Python-RT24.1/lib/python3.11/site-packages (from pandas<2.3.0,>=0.24.2->ibm-watsonx-ai) (2024.1)\nRequirement already satisfied: tzdata>=2022.1 in /opt/conda/envs/Python-RT24.1/lib/python3.11/site-packages (from pandas<2.3.0,>=0.24.2->ibm-watsonx-ai) (2023.3)\nRequirement already satisfied: charset-normalizer<4,>=2 in /opt/conda/envs/Python-RT24.1/lib/python3.11/site-packages (from requests->ibm-watsonx-ai) (2.0.4)\nRequirement already satisfied: six>=1.10.0 in /opt/conda/envs/Python-RT24.1/lib/python3.11/site-packages (from lomond->ibm-watsonx-ai) (1.16.0)\nRequirement already satisfied: sniffio>=1.1 in /opt/conda/envs/Python-RT24.1/lib/python3.11/site-packages (from anyio->httpx<0.29,>=0.27->ibm-watsonx-ai) (1.3.0)\n", "output_type": "stream"}, {"traceback": ["\u001b[0;31m---------------------------------------------------------------------------\u001b[0m", "\u001b[0;31mTypeError\u001b[0m                                 Traceback (most recent call last)", "Cell \u001b[0;32mIn[1], line 8\u001b[0m\n\u001b[1;32m      2\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mibm_watsonx_ai\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mfoundation_models\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m Model\n\u001b[1;32m      3\u001b[0m parameters \u001b[38;5;241m=\u001b[39m {\n\u001b[1;32m      4\u001b[0m     \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mdecoding_method\u001b[39m\u001b[38;5;124m\"\u001b[39m: \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mgreedy\u001b[39m\u001b[38;5;124m\"\u001b[39m,\n\u001b[1;32m      5\u001b[0m     \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mmax_new_tokens\u001b[39m\u001b[38;5;124m\"\u001b[39m: \u001b[38;5;241m100\u001b[39m\n\u001b[1;32m      6\u001b[0m }\n\u001b[0;32m----> 8\u001b[0m model \u001b[38;5;241m=\u001b[39m Model(\n\u001b[1;32m      9\u001b[0m     model_id\u001b[38;5;241m=\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mgranite-13b-instruct-v2\u001b[39m\u001b[38;5;124m\"\u001b[39m,\n\u001b[1;32m     10\u001b[0m     params\u001b[38;5;241m=\u001b[39mparameters,\n\u001b[1;32m     11\u001b[0m     project_id\u001b[38;5;241m=\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mce7c019d-e903-4766-aee3-b3e3c15574c9\u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m     12\u001b[0m )\n\u001b[1;32m     13\u001b[0m prompt \u001b[38;5;241m=\u001b[39m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mExplain the concept of AI to a 5-year-old kid.\u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m     15\u001b[0m response \u001b[38;5;241m=\u001b[39m model\u001b[38;5;241m.\u001b[39mgenerate(prompt)\n", "\u001b[0;31mTypeError\u001b[0m: Model.__init__() missing 1 required positional argument: 'credentials'"], "ename": "TypeError", "evalue": "Model.__init__() missing 1 required positional argument: 'credentials'", "output_type": "error"}], "execution_count": 1}, {"cell_type": "code", "source": "credentials = {\n    \"url\": \"https://eu-de.ml.cloud.ibm.com\",  # because you are using Frankfurt region\n    \"apikey\": \"2-_Znw1m23l5wpIA38xc2NElPo4UC6L79JqnuJTxNoY3\"\n}\nfrom ibm_watsonx_ai.foundation_models import Model\n\nparameters = {\n    \"decoding_method\": \"greedy\",\n    \"max_new_tokens\": 100\n}\n\nmodel = Model(\n    model_id=\"granite-13b-instruct-v2\",\n    params=parameters,\n    project_id=\"ce7c019d-e903-4766-aee3-b3e3c15574c9\",\n    credentials=credentials   # This is the missing argument you got an error for earlier\n)\n\nprompt = \"Explain the concept of AI to a 5-year-old kid.\"\nresponse = model.generate(prompt)\nprint(\"Generated Text:\", response)", "metadata": {"id": "a0c7ff32-79cd-4d12-8ba3-fd6bab2fffa5"}, "outputs": [{"name": "stderr", "text": "/opt/conda/envs/Python-RT24.1/lib/python3.11/site-packages/ibm_watsonx_ai/foundation_models/model.py:105: DeprecationWarning: The `Model` class is deprecated and will be removed in a future release. Please use the `ModelInference` class instead. To update your imports, use: `from ibm_watsonx_ai.foundation_models import ModelInference`.\n  warn(model_class_deprecated_warning, category=DeprecationWarning)\nFailure during Get available foundation models. (GET https://eu-de.ml.cloud.ibm.com/ml/v1/foundation_model_specs?version=2025-06-11&project_id=ce7c019d-e903-4766-aee3-b3e3c15574c9&filters=function_text_generation%2C%21lifecycle_withdrawn%3Aand&limit=200)\nStatus code: 403, body: {\"errors\":[{\"code\":\"no_associated_service_instance_error\",\"message\":\"project_id ce7c019d-e903-4766-aee3-b3e3c15574c9 is not associated with a WML instance\",\"more_info\":\"https://cloud.ibm.com/apidocs/watsonx-ai#list-foundation-model-specs\"}],\"trace\":\"04b399bb16b96c3e931c0c6502384faf\",\"status_code\":403}\nUnable to get model specifications from url: https://eu-de.ml.cloud.ibm.com\nReason: Failure during Get available foundation models. (GET https://eu-de.ml.cloud.ibm.com/ml/v1/foundation_model_specs?version=2025-06-11&project_id=ce7c019d-e903-4766-aee3-b3e3c15574c9&filters=function_text_generation%2C%21lifecycle_withdrawn%3Aand&limit=200)\nStatus code: 403, body: {\"errors\":[{\"code\":\"no_associated_service_instance_error\",\"message\":\"project_id ce7c019d-e903-4766-aee3-b3e3c15574c9 is not associated with a WML instance\",\"more_info\":\"https://cloud.ibm.com/apidocs/watsonx-ai#list-foundation-model-specs\"}],\"trace\":\"04b399bb16b96c3e931c0c6502384faf\",\"status_code\":403}\n", "output_type": "stream"}, {"traceback": ["\u001b[0;31m---------------------------------------------------------------------------\u001b[0m", "\u001b[0;31mApiRequestFailure\u001b[0m                         Traceback (most recent call last)", "File \u001b[0;32m/opt/conda/envs/Python-RT24.1/lib/python3.11/site-packages/ibm_watsonx_ai/foundation_models_manager.py:94\u001b[0m, in \u001b[0;36mFoundationModelsManager._get_spec\u001b[0;34m(self, url, operation_name, error_msg_id, model_id, limit, filters, asynchronous, get_all, tech_preview)\u001b[0m\n\u001b[1;32m     93\u001b[0m     \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[0;32m---> 94\u001b[0m         \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_get_with_or_without_limit(\n\u001b[1;32m     95\u001b[0m             url\u001b[38;5;241m=\u001b[39murl,\n\u001b[1;32m     96\u001b[0m             limit\u001b[38;5;241m=\u001b[39mlimit,\n\u001b[1;32m     97\u001b[0m             op_name\u001b[38;5;241m=\u001b[39moperation_name,\n\u001b[1;32m     98\u001b[0m             query_params\u001b[38;5;241m=\u001b[39mparams,\n\u001b[1;32m     99\u001b[0m             _async\u001b[38;5;241m=\u001b[39masynchronous,\n\u001b[1;32m    100\u001b[0m             _all\u001b[38;5;241m=\u001b[39mget_all,\n\u001b[1;32m    101\u001b[0m             skip_space_project_chk\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mTrue\u001b[39;00m,\n\u001b[1;32m    102\u001b[0m         )\n\u001b[1;32m    103\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m WMLClientError \u001b[38;5;28;01mas\u001b[39;00m e:\n", "File \u001b[0;32m/opt/conda/envs/Python-RT24.1/lib/python3.11/site-packages/ibm_watsonx_ai/wml_resource.py:535\u001b[0m, in \u001b[0;36mWMLResource._get_with_or_without_limit\u001b[0;34m(self, url, limit, op_name, summary, pre_defined, revision, skip_space_project_chk, query_params, _async, _all, _filter_func, _silent_response_logging)\u001b[0m\n\u001b[1;32m    527\u001b[0m response_get \u001b[38;5;241m=\u001b[39m request_handler\u001b[38;5;241m.\u001b[39mget(\n\u001b[1;32m    528\u001b[0m     url\u001b[38;5;241m=\u001b[39murl,\n\u001b[1;32m    529\u001b[0m     headers\u001b[38;5;241m=\u001b[39m\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_client\u001b[38;5;241m.\u001b[39m_get_headers(),\n\u001b[1;32m    530\u001b[0m     params\u001b[38;5;241m=\u001b[39mparams,\n\u001b[1;32m    531\u001b[0m )\n\u001b[1;32m    533\u001b[0m result \u001b[38;5;241m=\u001b[39m cast(\n\u001b[1;32m    534\u001b[0m     \u001b[38;5;28mdict\u001b[39m,\n\u001b[0;32m--> 535\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_handle_response(\n\u001b[1;32m    536\u001b[0m         \u001b[38;5;241m200\u001b[39m,\n\u001b[1;32m    537\u001b[0m         op_name,\n\u001b[1;32m    538\u001b[0m         response_get,\n\u001b[1;32m    539\u001b[0m         _silent_response_logging\u001b[38;5;241m=\u001b[39m_silent_response_logging,\n\u001b[1;32m    540\u001b[0m     ),\n\u001b[1;32m    541\u001b[0m )\n\u001b[1;32m    542\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mresources\u001b[39m\u001b[38;5;124m\"\u001b[39m \u001b[38;5;129;01min\u001b[39;00m result:\n", "File \u001b[0;32m/opt/conda/envs/Python-RT24.1/lib/python3.11/site-packages/ibm_watsonx_ai/wml_resource.py:167\u001b[0m, in \u001b[0;36mWMLResource._handle_response\u001b[0;34m(self, expected_status_code, operationName, response, json_response, _silent_response_logging, _field_to_hide)\u001b[0m\n\u001b[1;32m    166\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[0;32m--> 167\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m ApiRequestFailure(\n\u001b[1;32m    168\u001b[0m         \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mFailure during \u001b[39m\u001b[38;5;132;01m{}\u001b[39;00m\u001b[38;5;124m.\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;241m.\u001b[39mformat(operationName),\n\u001b[1;32m    169\u001b[0m         response,\n\u001b[1;32m    170\u001b[0m     )\n", "\u001b[0;31mApiRequestFailure\u001b[0m: Failure during Get available foundation models. (GET https://eu-de.ml.cloud.ibm.com/ml/v1/foundation_model_specs?version=2025-06-11&project_id=ce7c019d-e903-4766-aee3-b3e3c15574c9&filters=function_text_generation%2C%21lifecycle_withdrawn%3Aand&limit=200)\nStatus code: 403, body: {\"errors\":[{\"code\":\"no_associated_service_instance_error\",\"message\":\"project_id ce7c019d-e903-4766-aee3-b3e3c15574c9 is not associated with a WML instance\",\"more_info\":\"https://cloud.ibm.com/apidocs/watsonx-ai#list-foundation-model-specs\"}],\"trace\":\"04b399bb16b96c3e931c0c6502384faf\",\"status_code\":403}", "\nDuring handling of the above exception, another exception occurred:\n", "\u001b[0;31mWMLClientError\u001b[0m                            Traceback (most recent call last)", "Cell \u001b[0;32mIn[2], line 12\u001b[0m\n\u001b[1;32m      5\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mibm_watsonx_ai\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mfoundation_models\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m Model\n\u001b[1;32m      7\u001b[0m parameters \u001b[38;5;241m=\u001b[39m {\n\u001b[1;32m      8\u001b[0m     \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mdecoding_method\u001b[39m\u001b[38;5;124m\"\u001b[39m: \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mgreedy\u001b[39m\u001b[38;5;124m\"\u001b[39m,\n\u001b[1;32m      9\u001b[0m     \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mmax_new_tokens\u001b[39m\u001b[38;5;124m\"\u001b[39m: \u001b[38;5;241m100\u001b[39m\n\u001b[1;32m     10\u001b[0m }\n\u001b[0;32m---> 12\u001b[0m model \u001b[38;5;241m=\u001b[39m Model(\n\u001b[1;32m     13\u001b[0m     model_id\u001b[38;5;241m=\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mgranite-13b-instruct-v2\u001b[39m\u001b[38;5;124m\"\u001b[39m,\n\u001b[1;32m     14\u001b[0m     params\u001b[38;5;241m=\u001b[39mparameters,\n\u001b[1;32m     15\u001b[0m     project_id\u001b[38;5;241m=\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mce7c019d-e903-4766-aee3-b3e3c15574c9\u001b[39m\u001b[38;5;124m\"\u001b[39m,\n\u001b[1;32m     16\u001b[0m     credentials\u001b[38;5;241m=\u001b[39mcredentials   \u001b[38;5;66;03m# This is the missing argument you got an error for earlier\u001b[39;00m\n\u001b[1;32m     17\u001b[0m )\n\u001b[1;32m     19\u001b[0m prompt \u001b[38;5;241m=\u001b[39m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mExplain the concept of AI to a 5-year-old kid.\u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m     20\u001b[0m response \u001b[38;5;241m=\u001b[39m model\u001b[38;5;241m.\u001b[39mgenerate(prompt)\n", "File \u001b[0;32m/opt/conda/envs/Python-RT24.1/lib/python3.11/site-packages/ibm_watsonx_ai/foundation_models/model.py:107\u001b[0m, in \u001b[0;36mModel.__init__\u001b[0;34m(self, model_id, credentials, params, project_id, space_id, verify, validate)\u001b[0m\n\u001b[1;32m    100\u001b[0m     model_class_deprecated_warning \u001b[38;5;241m=\u001b[39m (\n\u001b[1;32m    101\u001b[0m         \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mThe `Model` class is deprecated and will be removed in a future release. \u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m    102\u001b[0m         \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mPlease use the `ModelInference` class instead. \u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m    103\u001b[0m         \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mTo update your imports, use: `from ibm_watsonx_ai.foundation_models import ModelInference`.\u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m    104\u001b[0m     )\n\u001b[1;32m    105\u001b[0m     warn(model_class_deprecated_warning, category\u001b[38;5;241m=\u001b[39m\u001b[38;5;167;01mDeprecationWarning\u001b[39;00m)\n\u001b[0;32m--> 107\u001b[0m ModelInference\u001b[38;5;241m.\u001b[39m\u001b[38;5;21m__init__\u001b[39m(\n\u001b[1;32m    108\u001b[0m     \u001b[38;5;28mself\u001b[39m,\n\u001b[1;32m    109\u001b[0m     model_id\u001b[38;5;241m=\u001b[39mmodel_id,\n\u001b[1;32m    110\u001b[0m     credentials\u001b[38;5;241m=\u001b[39mcredentials,\n\u001b[1;32m    111\u001b[0m     params\u001b[38;5;241m=\u001b[39mparams,\n\u001b[1;32m    112\u001b[0m     project_id\u001b[38;5;241m=\u001b[39mproject_id,\n\u001b[1;32m    113\u001b[0m     space_id\u001b[38;5;241m=\u001b[39mspace_id,\n\u001b[1;32m    114\u001b[0m     verify\u001b[38;5;241m=\u001b[39mverify,\n\u001b[1;32m    115\u001b[0m     validate\u001b[38;5;241m=\u001b[39mvalidate,\n\u001b[1;32m    116\u001b[0m )\n", "File \u001b[0;32m/opt/conda/envs/Python-RT24.1/lib/python3.11/site-packages/ibm_watsonx_ai/foundation_models/inference/model_inference.py:222\u001b[0m, in \u001b[0;36mModelInference.__init__\u001b[0;34m(self, model_id, deployment_id, params, credentials, project_id, space_id, verify, api_client, validate, persistent_connection, max_retries, delay_time, retry_status_codes)\u001b[0m\n\u001b[1;32m    220\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_inference: BaseModelInference\n\u001b[1;32m    221\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mmodel_id:\n\u001b[0;32m--> 222\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_inference \u001b[38;5;241m=\u001b[39m FMModelInference(\n\u001b[1;32m    223\u001b[0m         model_id\u001b[38;5;241m=\u001b[39m\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mmodel_id,\n\u001b[1;32m    224\u001b[0m         api_client\u001b[38;5;241m=\u001b[39m\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_client,\n\u001b[1;32m    225\u001b[0m         params\u001b[38;5;241m=\u001b[39m\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mparams,\n\u001b[1;32m    226\u001b[0m         validate\u001b[38;5;241m=\u001b[39mvalidate,\n\u001b[1;32m    227\u001b[0m         persistent_connection\u001b[38;5;241m=\u001b[39mpersistent_connection,\n\u001b[1;32m    228\u001b[0m         max_retries\u001b[38;5;241m=\u001b[39mmax_retries,\n\u001b[1;32m    229\u001b[0m         delay_time\u001b[38;5;241m=\u001b[39mdelay_time,\n\u001b[1;32m    230\u001b[0m         retry_status_codes\u001b[38;5;241m=\u001b[39mretry_status_codes,\n\u001b[1;32m    231\u001b[0m     )\n\u001b[1;32m    232\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m    233\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mdeployment_id \u001b[38;5;241m=\u001b[39m cast(\u001b[38;5;28mstr\u001b[39m, \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mdeployment_id)\n", "File \u001b[0;32m/opt/conda/envs/Python-RT24.1/lib/python3.11/site-packages/ibm_watsonx_ai/foundation_models/inference/fm_model_inference.py:71\u001b[0m, in \u001b[0;36mFMModelInference.__init__\u001b[0;34m(self, model_id, api_client, params, validate, persistent_connection, max_retries, delay_time, retry_status_codes)\u001b[0m\n\u001b[1;32m     68\u001b[0m model_specs: \u001b[38;5;28mdict\u001b[39m \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mdict\u001b[39m()\n\u001b[1;32m     70\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_client\u001b[38;5;241m.\u001b[39m_use_fm_ga_api:\n\u001b[0;32m---> 71\u001b[0m     model_specs \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_client\u001b[38;5;241m.\u001b[39mfoundation_models\u001b[38;5;241m.\u001b[39mget_model_specs()  \u001b[38;5;66;03m# type: ignore[assignment]\u001b[39;00m\n\u001b[1;32m     72\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m     73\u001b[0m     \u001b[38;5;28;01mwith\u001b[39;00m catch_warnings():\n", "File \u001b[0;32m/opt/conda/envs/Python-RT24.1/lib/python3.11/site-packages/ibm_watsonx_ai/foundation_models_manager.py:148\u001b[0m, in \u001b[0;36mFoundationModelsManager.get_model_specs\u001b[0;34m(self, model_id, limit, asynchronous, get_all, **kwargs)\u001b[0m\n\u001b[1;32m    112\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mget_model_specs\u001b[39m(\n\u001b[1;32m    113\u001b[0m     \u001b[38;5;28mself\u001b[39m,\n\u001b[1;32m    114\u001b[0m     model_id: \u001b[38;5;28mstr\u001b[39m \u001b[38;5;241m|\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m,\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m    118\u001b[0m     \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mkwargs: Any,\n\u001b[1;32m    119\u001b[0m ) \u001b[38;5;241m-\u001b[39m\u001b[38;5;241m>\u001b[39m \u001b[38;5;28mdict\u001b[39m \u001b[38;5;241m|\u001b[39m Generator \u001b[38;5;241m|\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m:\n\u001b[1;32m    120\u001b[0m \u001b[38;5;250m    \u001b[39m\u001b[38;5;124;03m\"\"\"\u001b[39;00m\n\u001b[1;32m    121\u001b[0m \u001b[38;5;124;03m    Retrieves a list of specifications for a deployed foundation model.\u001b[39;00m\n\u001b[1;32m    122\u001b[0m \n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m    146\u001b[0m \u001b[38;5;124;03m        client.foundation_models.get_model_specs(model_id=\"google/flan-ul2\")\u001b[39;00m\n\u001b[1;32m    147\u001b[0m \u001b[38;5;124;03m    \"\"\"\u001b[39;00m\n\u001b[0;32m--> 148\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_get_spec(\n\u001b[1;32m    149\u001b[0m         url\u001b[38;5;241m=\u001b[39m\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_client\u001b[38;5;241m.\u001b[39m_href_definitions\u001b[38;5;241m.\u001b[39mget_fm_specifications_href(),\n\u001b[1;32m    150\u001b[0m         operation_name\u001b[38;5;241m=\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mGet available foundation models\u001b[39m\u001b[38;5;124m\"\u001b[39m,\n\u001b[1;32m    151\u001b[0m         error_msg_id\u001b[38;5;241m=\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mfm_prompt_tuning_no_model_specs\u001b[39m\u001b[38;5;124m\"\u001b[39m,\n\u001b[1;32m    152\u001b[0m         filters\u001b[38;5;241m=\u001b[39m(\n\u001b[1;32m    153\u001b[0m             \u001b[38;5;28;01mNone\u001b[39;00m\n\u001b[1;32m    154\u001b[0m             \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_client\u001b[38;5;241m.\u001b[39mCPD_version \u001b[38;5;241m<\u001b[39m \u001b[38;5;241m5.0\u001b[39m\n\u001b[1;32m    155\u001b[0m             \u001b[38;5;28;01melse\u001b[39;00m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mfunction_text_generation,!lifecycle_withdrawn:and\u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m    156\u001b[0m         ),\n\u001b[1;32m    157\u001b[0m         model_id\u001b[38;5;241m=\u001b[39mmodel_id,\n\u001b[1;32m    158\u001b[0m         limit\u001b[38;5;241m=\u001b[39mlimit,\n\u001b[1;32m    159\u001b[0m         asynchronous\u001b[38;5;241m=\u001b[39masynchronous,\n\u001b[1;32m    160\u001b[0m         get_all\u001b[38;5;241m=\u001b[39mget_all,\n\u001b[1;32m    161\u001b[0m         tech_preview\u001b[38;5;241m=\u001b[39mkwargs\u001b[38;5;241m.\u001b[39mget(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mtech_preview\u001b[39m\u001b[38;5;124m\"\u001b[39m, \u001b[38;5;28;01mFalse\u001b[39;00m),\n\u001b[1;32m    162\u001b[0m     )\n", "File \u001b[0;32m/opt/conda/envs/Python-RT24.1/lib/python3.11/site-packages/ibm_watsonx_ai/foundation_models_manager.py:104\u001b[0m, in \u001b[0;36mFoundationModelsManager._get_spec\u001b[0;34m(self, url, operation_name, error_msg_id, model_id, limit, filters, asynchronous, get_all, tech_preview)\u001b[0m\n\u001b[1;32m     94\u001b[0m         \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_get_with_or_without_limit(\n\u001b[1;32m     95\u001b[0m             url\u001b[38;5;241m=\u001b[39murl,\n\u001b[1;32m     96\u001b[0m             limit\u001b[38;5;241m=\u001b[39mlimit,\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m    101\u001b[0m             skip_space_project_chk\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mTrue\u001b[39;00m,\n\u001b[1;32m    102\u001b[0m         )\n\u001b[1;32m    103\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m WMLClientError \u001b[38;5;28;01mas\u001b[39;00m e:\n\u001b[0;32m--> 104\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m WMLClientError(\n\u001b[1;32m    105\u001b[0m         Messages\u001b[38;5;241m.\u001b[39mget_message(\n\u001b[1;32m    106\u001b[0m             \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_client\u001b[38;5;241m.\u001b[39mcredentials\u001b[38;5;241m.\u001b[39murl,\n\u001b[1;32m    107\u001b[0m             message_id\u001b[38;5;241m=\u001b[39merror_msg_id,\n\u001b[1;32m    108\u001b[0m         ),\n\u001b[1;32m    109\u001b[0m         e,\n\u001b[1;32m    110\u001b[0m     )\n", "\u001b[0;31mWMLClientError\u001b[0m: Unable to get model specifications from url: https://eu-de.ml.cloud.ibm.com\nReason: Failure during Get available foundation models. (GET https://eu-de.ml.cloud.ibm.com/ml/v1/foundation_model_specs?version=2025-06-11&project_id=ce7c019d-e903-4766-aee3-b3e3c15574c9&filters=function_text_generation%2C%21lifecycle_withdrawn%3Aand&limit=200)\nStatus code: 403, body: {\"errors\":[{\"code\":\"no_associated_service_instance_error\",\"message\":\"project_id ce7c019d-e903-4766-aee3-b3e3c15574c9 is not associated with a WML instance\",\"more_info\":\"https://cloud.ibm.com/apidocs/watsonx-ai#list-foundation-model-specs\"}],\"trace\":\"04b399bb16b96c3e931c0c6502384faf\",\"status_code\":403}"], "ename": "WMLClientError", "evalue": "Unable to get model specifications from url: https://eu-de.ml.cloud.ibm.com\nReason: Failure during Get available foundation models. (GET https://eu-de.ml.cloud.ibm.com/ml/v1/foundation_model_specs?version=2025-06-11&project_id=ce7c019d-e903-4766-aee3-b3e3c15574c9&filters=function_text_generation%2C%21lifecycle_withdrawn%3Aand&limit=200)\nStatus code: 403, body: {\"errors\":[{\"code\":\"no_associated_service_instance_error\",\"message\":\"project_id ce7c019d-e903-4766-aee3-b3e3c15574c9 is not associated with a WML instance\",\"more_info\":\"https://cloud.ibm.com/apidocs/watsonx-ai#list-foundation-model-specs\"}],\"trace\":\"04b399bb16b96c3e931c0c6502384faf\",\"status_code\":403}", "output_type": "error"}], "execution_count": 2}, {"cell_type": "code", "source": "credentials = {\n    \"url\": \"https://eu-de.ml.cloud.ibm.com\",  # because you are using Frankfurt region\n    \"apikey\": \"2-_Znw1m23l5wpIA38xc2NElPo4UC6L79JqnuJTxNoY3\"\n}\nfrom ibm_watsonx_ai.foundation_models import Model\n\nparameters = {\n    \"decoding_method\": \"greedy\",\n    \"max_new_tokens\": 100\n}\n\nmodel = Model(\n    model_id=\"granite-13b-instruct-v2\",\n    params=parameters,\n    project_id=\"ce7c019d-e903-4766-aee3-b3e3c15574c9\",\n    credentials=credentials   # This is the missing argument you got an error for earlier\n)\n\nprompt = \"Explain the concept of AI to a 5-year-old kid.\"\nresponse = model.generate(prompt)\nprint(\"Generated Text:\", response)", "metadata": {"id": "c39d6103-906f-4959-86ad-8b5bc328e391"}, "outputs": [{"name": "stderr", "text": "/opt/conda/envs/Python-RT24.1/lib/python3.11/site-packages/ibm_watsonx_ai/foundation_models/model.py:105: DeprecationWarning: The `Model` class is deprecated and will be removed in a future release. Please use the `ModelInference` class instead. To update your imports, use: `from ibm_watsonx_ai.foundation_models import ModelInference`.\n  warn(model_class_deprecated_warning, category=DeprecationWarning)\nModel 'granite-13b-instruct-v2' is not supported for this environment. Supported models: ['core42/jais-13b-chat', 'google/flan-t5-xl', 'google/flan-t5-xxl', 'google/flan-ul2', 'ibm/granite-13b-instruct-v2', 'ibm/granite-20b-code-instruct', 'ibm/granite-3-2b-instruct', 'ibm/granite-3-3-8b-instruct', 'ibm/granite-3-8b-instruct', 'ibm/granite-34b-code-instruct', 'meta-llama/llama-2-13b-chat', 'meta-llama/llama-3-2-11b-vision-instruct', 'meta-llama/llama-3-2-90b-vision-instruct', 'meta-llama/llama-3-3-70b-instruct', 'meta-llama/llama-4-maverick-17b-128e-instruct-fp8', 'mistralai/mistral-large', 'mistralai/mistral-small-24b-instruct-2501', 'mistralai/mistral-small-3-1-24b-instruct-2503', 'mistralai/mixtral-8x7b-instruct-v01', 'mistralai/pixtral-12b', 'sdaia/allam-1-13b-instruct']\n", "output_type": "stream"}, {"traceback": ["\u001b[0;31m---------------------------------------------------------------------------\u001b[0m", "\u001b[0;31mWMLClientError\u001b[0m                            Traceback (most recent call last)", "Cell \u001b[0;32mIn[1], line 12\u001b[0m\n\u001b[1;32m      5\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mibm_watsonx_ai\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mfoundation_models\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m Model\n\u001b[1;32m      7\u001b[0m parameters \u001b[38;5;241m=\u001b[39m {\n\u001b[1;32m      8\u001b[0m     \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mdecoding_method\u001b[39m\u001b[38;5;124m\"\u001b[39m: \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mgreedy\u001b[39m\u001b[38;5;124m\"\u001b[39m,\n\u001b[1;32m      9\u001b[0m     \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mmax_new_tokens\u001b[39m\u001b[38;5;124m\"\u001b[39m: \u001b[38;5;241m100\u001b[39m\n\u001b[1;32m     10\u001b[0m }\n\u001b[0;32m---> 12\u001b[0m model \u001b[38;5;241m=\u001b[39m Model(\n\u001b[1;32m     13\u001b[0m     model_id\u001b[38;5;241m=\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mgranite-13b-instruct-v2\u001b[39m\u001b[38;5;124m\"\u001b[39m,\n\u001b[1;32m     14\u001b[0m     params\u001b[38;5;241m=\u001b[39mparameters,\n\u001b[1;32m     15\u001b[0m     project_id\u001b[38;5;241m=\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mce7c019d-e903-4766-aee3-b3e3c15574c9\u001b[39m\u001b[38;5;124m\"\u001b[39m,\n\u001b[1;32m     16\u001b[0m     credentials\u001b[38;5;241m=\u001b[39mcredentials   \u001b[38;5;66;03m# This is the missing argument you got an error for earlier\u001b[39;00m\n\u001b[1;32m     17\u001b[0m )\n\u001b[1;32m     19\u001b[0m prompt \u001b[38;5;241m=\u001b[39m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mExplain the concept of AI to a 5-year-old kid.\u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m     20\u001b[0m response \u001b[38;5;241m=\u001b[39m model\u001b[38;5;241m.\u001b[39mgenerate(prompt)\n", "File \u001b[0;32m/opt/conda/envs/Python-RT24.1/lib/python3.11/site-packages/ibm_watsonx_ai/foundation_models/model.py:107\u001b[0m, in \u001b[0;36mModel.__init__\u001b[0;34m(self, model_id, credentials, params, project_id, space_id, verify, validate)\u001b[0m\n\u001b[1;32m    100\u001b[0m     model_class_deprecated_warning \u001b[38;5;241m=\u001b[39m (\n\u001b[1;32m    101\u001b[0m         \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mThe `Model` class is deprecated and will be removed in a future release. \u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m    102\u001b[0m         \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mPlease use the `ModelInference` class instead. \u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m    103\u001b[0m         \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mTo update your imports, use: `from ibm_watsonx_ai.foundation_models import ModelInference`.\u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m    104\u001b[0m     )\n\u001b[1;32m    105\u001b[0m     warn(model_class_deprecated_warning, category\u001b[38;5;241m=\u001b[39m\u001b[38;5;167;01mDeprecationWarning\u001b[39;00m)\n\u001b[0;32m--> 107\u001b[0m ModelInference\u001b[38;5;241m.\u001b[39m\u001b[38;5;21m__init__\u001b[39m(\n\u001b[1;32m    108\u001b[0m     \u001b[38;5;28mself\u001b[39m,\n\u001b[1;32m    109\u001b[0m     model_id\u001b[38;5;241m=\u001b[39mmodel_id,\n\u001b[1;32m    110\u001b[0m     credentials\u001b[38;5;241m=\u001b[39mcredentials,\n\u001b[1;32m    111\u001b[0m     params\u001b[38;5;241m=\u001b[39mparams,\n\u001b[1;32m    112\u001b[0m     project_id\u001b[38;5;241m=\u001b[39mproject_id,\n\u001b[1;32m    113\u001b[0m     space_id\u001b[38;5;241m=\u001b[39mspace_id,\n\u001b[1;32m    114\u001b[0m     verify\u001b[38;5;241m=\u001b[39mverify,\n\u001b[1;32m    115\u001b[0m     validate\u001b[38;5;241m=\u001b[39mvalidate,\n\u001b[1;32m    116\u001b[0m )\n", "File \u001b[0;32m/opt/conda/envs/Python-RT24.1/lib/python3.11/site-packages/ibm_watsonx_ai/foundation_models/inference/model_inference.py:222\u001b[0m, in \u001b[0;36mModelInference.__init__\u001b[0;34m(self, model_id, deployment_id, params, credentials, project_id, space_id, verify, api_client, validate, persistent_connection, max_retries, delay_time, retry_status_codes)\u001b[0m\n\u001b[1;32m    220\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_inference: BaseModelInference\n\u001b[1;32m    221\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mmodel_id:\n\u001b[0;32m--> 222\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_inference \u001b[38;5;241m=\u001b[39m FMModelInference(\n\u001b[1;32m    223\u001b[0m         model_id\u001b[38;5;241m=\u001b[39m\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mmodel_id,\n\u001b[1;32m    224\u001b[0m         api_client\u001b[38;5;241m=\u001b[39m\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_client,\n\u001b[1;32m    225\u001b[0m         params\u001b[38;5;241m=\u001b[39m\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mparams,\n\u001b[1;32m    226\u001b[0m         validate\u001b[38;5;241m=\u001b[39mvalidate,\n\u001b[1;32m    227\u001b[0m         persistent_connection\u001b[38;5;241m=\u001b[39mpersistent_connection,\n\u001b[1;32m    228\u001b[0m         max_retries\u001b[38;5;241m=\u001b[39mmax_retries,\n\u001b[1;32m    229\u001b[0m         delay_time\u001b[38;5;241m=\u001b[39mdelay_time,\n\u001b[1;32m    230\u001b[0m         retry_status_codes\u001b[38;5;241m=\u001b[39mretry_status_codes,\n\u001b[1;32m    231\u001b[0m     )\n\u001b[1;32m    232\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m    233\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mdeployment_id \u001b[38;5;241m=\u001b[39m cast(\u001b[38;5;28mstr\u001b[39m, \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mdeployment_id)\n", "File \u001b[0;32m/opt/conda/envs/Python-RT24.1/lib/python3.11/site-packages/ibm_watsonx_ai/foundation_models/inference/fm_model_inference.py:101\u001b[0m, in \u001b[0;36mFMModelInference.__init__\u001b[0;34m(self, model_id, api_client, params, validate, persistent_connection, max_retries, delay_time, retry_status_codes)\u001b[0m\n\u001b[1;32m     98\u001b[0m                 \u001b[38;5;28;01mbreak\u001b[39;00m\n\u001b[1;32m    100\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_tech_preview:\n\u001b[0;32m--> 101\u001b[0m         \u001b[38;5;28;01mraise\u001b[39;00m WMLClientError(\n\u001b[1;32m    102\u001b[0m             error_msg\u001b[38;5;241m=\u001b[39m\u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mModel \u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;132;01m{\u001b[39;00m\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mmodel_id\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124m is not supported for this environment. \u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m    103\u001b[0m             \u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mSupported models: \u001b[39m\u001b[38;5;132;01m{\u001b[39;00msupported_models\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m    104\u001b[0m         )\n\u001b[1;32m    106\u001b[0m \u001b[38;5;66;03m# check if model is in constricted mode\u001b[39;00m\n\u001b[1;32m    107\u001b[0m _check_model_state(\n\u001b[1;32m    108\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_client,\n\u001b[1;32m    109\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mmodel_id,\n\u001b[1;32m    110\u001b[0m     tech_preview\u001b[38;5;241m=\u001b[39m\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_tech_preview,\n\u001b[1;32m    111\u001b[0m     model_specs\u001b[38;5;241m=\u001b[39mmodel_specs,\n\u001b[1;32m    112\u001b[0m )\n", "\u001b[0;31mWMLClientError\u001b[0m: Model 'granite-13b-instruct-v2' is not supported for this environment. Supported models: ['core42/jais-13b-chat', 'google/flan-t5-xl', 'google/flan-t5-xxl', 'google/flan-ul2', 'ibm/granite-13b-instruct-v2', 'ibm/granite-20b-code-instruct', 'ibm/granite-3-2b-instruct', 'ibm/granite-3-3-8b-instruct', 'ibm/granite-3-8b-instruct', 'ibm/granite-34b-code-instruct', 'meta-llama/llama-2-13b-chat', 'meta-llama/llama-3-2-11b-vision-instruct', 'meta-llama/llama-3-2-90b-vision-instruct', 'meta-llama/llama-3-3-70b-instruct', 'meta-llama/llama-4-maverick-17b-128e-instruct-fp8', 'mistralai/mistral-large', 'mistralai/mistral-small-24b-instruct-2501', 'mistralai/mistral-small-3-1-24b-instruct-2503', 'mistralai/mixtral-8x7b-instruct-v01', 'mistralai/pixtral-12b', 'sdaia/allam-1-13b-instruct']"], "ename": "WMLClientError", "evalue": "Model 'granite-13b-instruct-v2' is not supported for this environment. Supported models: ['core42/jais-13b-chat', 'google/flan-t5-xl', 'google/flan-t5-xxl', 'google/flan-ul2', 'ibm/granite-13b-instruct-v2', 'ibm/granite-20b-code-instruct', 'ibm/granite-3-2b-instruct', 'ibm/granite-3-3-8b-instruct', 'ibm/granite-3-8b-instruct', 'ibm/granite-34b-code-instruct', 'meta-llama/llama-2-13b-chat', 'meta-llama/llama-3-2-11b-vision-instruct', 'meta-llama/llama-3-2-90b-vision-instruct', 'meta-llama/llama-3-3-70b-instruct', 'meta-llama/llama-4-maverick-17b-128e-instruct-fp8', 'mistralai/mistral-large', 'mistralai/mistral-small-24b-instruct-2501', 'mistralai/mistral-small-3-1-24b-instruct-2503', 'mistralai/mixtral-8x7b-instruct-v01', 'mistralai/pixtral-12b', 'sdaia/allam-1-13b-instruct']", "output_type": "error"}], "execution_count": 1}, {"cell_type": "code", "source": "credentials = {\n    \"url\": \"https://eu-de.ml.cloud.ibm.com\",  # because you are using Frankfurt region\n    \"apikey\": \"2-_Znw1m23l5wpIA38xc2NElPo4UC6L79JqnuJTxNoY3\"\n}\nfrom ibm_watsonx_ai.foundation_models import Model\n\nparameters = {\n    \"decoding_method\": \"greedy\",\n    \"max_new_tokens\": 100\n}\n\nmodel = Model(\n    model_id=\"ibm/granite-13b-instruct-v2\",\n    params=parameters,\n    project_id=\"ce7c019d-e903-4766-aee3-b3e3c15574c9\",\n    credentials=credentials   # This is the missing argument you got an error for earlier\n)\n\nprompt = \"Explain the concept of AI to a 5-year-old kid.\"\nresponse = model.generate(prompt)\nprint(\"Generated Text:\", response)", "metadata": {"id": "6dfe0658-46dd-4a2d-85ed-9da9bae02eae"}, "outputs": [{"name": "stderr", "text": "/opt/conda/envs/Python-RT24.1/lib/python3.11/site-packages/ibm_watsonx_ai/foundation_models/model.py:105: DeprecationWarning: The `Model` class is deprecated and will be removed in a future release. Please use the `ModelInference` class instead. To update your imports, use: `from ibm_watsonx_ai.foundation_models import ModelInference`.\n  warn(model_class_deprecated_warning, category=DeprecationWarning)\n/opt/conda/envs/Python-RT24.1/lib/python3.11/site-packages/ibm_watsonx_ai/foundation_models/utils/utils.py:436: LifecycleWarning: Model 'ibm/granite-13b-instruct-v2' is in deprecated state from 2025-06-18 until 2025-10-15. IDs of alternative models: ibm/granite-3-3-8b-instruct. Further details: https://dataplatform.cloud.ibm.com/docs/content/wsj/analyze-data/fm-model-lifecycle.html?context=wx&audience=wdp\n  warn(model_state_warning, category=LifecycleWarning)\n", "output_type": "stream"}, {"name": "stdout", "text": "Generated Text: {'model_id': 'ibm/granite-13b-instruct-v2', 'created_at': '2025-06-19T11:42:57.485Z', 'results': [{'generated_text': 'Artificial intelligence is a branch of computer science that focuses on creating intelligent machines that can think and act like humans.', 'generated_token_count': 24, 'input_token_count': 15, 'stop_reason': 'eos_token'}], 'system': {'warnings': [{'message': \"Model 'ibm/granite-13b-instruct-v2' is in deprecated state from 2025-06-18. It will be in withdrawn state from 2025-10-15. IDs of alternative models: ibm/granite-3-3-8b-instruct.\", 'id': 'deprecation_warning', 'more_info': 'https://dataplatform.cloud.ibm.com/docs/content/wsj/analyze-data/fm-model-lifecycle.html?context=wx&audience=wdp'}, {'message': \"This API is deprecated. Please use '/text/chat' instead.\", 'id': 'api_deprecation'}]}}\n", "output_type": "stream"}], "execution_count": 2}, {"cell_type": "code", "source": "from ibm_watsonx_ai.foundation_models import ModelInference\n\nmodel_id = \"ibm/granite-3-8b-instruct\"  \nproject_id = \"ce7c019d-e903-4766-aee3-b3e3c15574c9\" \n\ncredentials = {\n    \"url\": \"https://eu-de.ml.cloud.ibm.com\",  \n    \"apikey\": \"2-_Znw1m23l5wpIA38xc2NElPo4UC6L79JqnuJTxNoY3\"  \n}\n\nmodel = ModelInference(\n    model_id=model_id,\n    params={\"decoding_method\": \"greedy\"},  \n    project_id=project_id,\n    credentials=credentials\n)\n\nprompt = \"Explain the concept of AI to a 5-year-old-kid\"\n\nresponse = model.generate(prompt)\n\nprint(\"Generated Text:\", response)", "metadata": {"id": "6432ee31-1d1e-44f7-a8b9-f51e792dfbdf"}, "outputs": [{"name": "stdout", "text": "Generated Text: {'model_id': 'ibm/granite-3-8b-instruct', 'model_version': '1.1.0', 'created_at': '2025-06-19T11:53:44.886Z', 'results': [{'generated_text': '.\\n\\nImagine if your toys could think and learn all by themselves, just like you', 'generated_token_count': 20, 'input_token_count': 15, 'stop_reason': 'max_tokens'}], 'system': {'warnings': [{'message': \"The value of 'parameters.max_new_tokens' for this model was set to value 20\", 'id': 'unspecified_max_new_tokens', 'additional_properties': {'limit': 0, 'new_value': 20, 'parameter': 'parameters.max_new_tokens', 'value': 0}}, {'message': \"This API is deprecated. Please use '/text/chat' instead.\", 'id': 'api_deprecation'}]}}\n", "output_type": "stream"}], "execution_count": 5}, {"cell_type": "code", "source": "", "metadata": {"id": "354438e9-52a3-452e-878a-edef744a62cd"}, "outputs": [], "execution_count": null}]}